### PR TITLE
config: turning down v1 config

### PR DESCRIPTION
**Pull request by alyssawilk**
I'm inclined to make the smallest possible change, removing the v1 enabling flag, and see if it sticks.

*Risk Level*: High
*Testing*: removed the v1 compat test
*Docs Changes*: not with this PR
*Release Notes*: Inline.
Part of #4617

### CODE REVIEW

**Comment by alyssawilk** [comment]
@envoyproxy/maintainers anyone have any final objections?

Once I think we're good to merge I'll send an email to envoy-announce and ship it.

File: source/server/options_impl.cc [hunk 1]
```diff
@@ -194,7 +191,7 @@ OptionsImpl::OptionsImpl(int argc, const char* const* argv,
   concurrency_ = std::max(1U, concurrency.getValue());
   config_path_ = config_path.getValue();
   config_yaml_ = config_yaml.getValue();
-  v2_config_only_ = !allow_v1_config.getValue();
+  v2_config_only_ = true;
```
Comment (mattklein123) [inline source/server/options_impl.cc]: You could probably kill this member variable now?

**Comment by lizan** [comment]
Changes LGTM. Should we kill v1 docs first? v1 docs appears everywhere and in higher place than v2 docs in most of search result...

File: docs/root/intro/version_history.rst [hunk 2]
```diff
@@ -28,7 +29,7 @@ Version history
   health check/weight/metadata updates within the given duration.
 * config: regex validation added to limit to a maximum of 1024 characters.
 * config: v1 disabled by default. v1 support remains available until October via flipping --v2-config-only=false.
-* config: v1 disabled by default. v1 support remains available until October via setting :option:`--allow-deprecated-v1-api`.
+* config: v1 disabled by default. v1 support remains available until October via deprecated flag --allow-deprecated-v1-api.
```
Comment (htuch) [inline docs/root/intro/version_history.rst]: I don't think we should modify earlier release notes, this is a bit like rewriting history. Ideally once we release, they are immutable.

Comment (alyssawilk) [reply docs/root/intro/version_history.rst]: Unfortunately I had to - the :option: tag was breaking the docs build now that the flag didn't exist.
I figured as long as I was removing the tagging I should explain why.

**Comment by alyssawilk** [comment]
I'm inclined to make sure the v1 turndown sticks, but I'm happy to start the docs PR "in parallel" since digging out all the v1 references is probably going to take some time.

File: docs/root/intro/version_history.rst (hunk 2 above)
Comment (htuch) [reply docs/root/intro/version_history.rst]: So be it.

**Pull request merged**